### DISCUSSION TITLE
Event.target.value is delayed one step in Next.js

### DISCUSSION THREAD

**Discussion by mihaimazareanu** [neuefische/questions#340] [Web] [answered]
Hi all,

for my capstone project I am trying to implement a password validation functionality when the user creates an account, meaning there are 2 input elements, "Choose password" and "Repeat password". In order to be able to compare the values from the 2 inputs and then validate (or not) creating the account, I am trying to get the values from the 2 input fields.

For that I created to state variables for each inout field:

    const [inputPassword, setInputPassword] = useState("");
    const [inputRepeatPassword, setInputRepeatPassword] = useState("");

Then I added for each input an onChange function:

    <InputChoosePassword
                  onChange={handleChangePassword}
                  type={showPassword ? "text" : "password"}
                  required
                  placeholder="Choose a safe password..."
                  pattern="(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{8,}"
                />

    <InputRepeatPassword
                type="password"
                placeholder="Repeat password..."
                required 
                onChange={() => handleChangeRepeatPassword()}
              />

InputChoosePassword and InputRepeatPassword are styled components of type input.

`type={showPassword ? "text" : "password"}` is for a show/hide password function, for which I used another state variable showPassword.

Then I created the handleChange functions for each input:

    const handleChangePassword = event => { 
    setInputPassword(event.target.value);
    console.log(inputPassword);
    };

    const handleChangeRepeatPassword = event => {
    setInputRepeatPassword(event.target.value);
    console.log(inputRepeatPassword);
    };

However, console.log shows that the 2 state variables `inputPassword` and `inputRepeatPassword` get updated with a one step delay.

Any idea how I might solve this issue?

**Comment by actyralf** [comment, marked as answer]
Hi Mihai,

state changes are NOT reflected immediately - when you console.log() a state value after setting ist you see the old value. Try to place the values somewhere on screen and you will see that they get updated correctly. 

However there seems to be a problem with your handler for the repeated password:

```
onChange={() => handleChangeRepeatPassword()}
```
If you call the handler this way, the event will not be passed to the handler.

**Reply by mihaimazareanu** [reply]
Thanks for the reply, I will look into calling the handler correctly then...

**Reply by mihaimazareanu** [reply]
Ok, I see what I did... I called the handleChange functions wrong on both inputs, then noticed it and changed it only in the first one. :( Thanks for the heads-up.
